<!-- Launch file for camera main on Rb5. -->

<launch>
  <arg name="camera_id" default="0" /> <!-- 0 for main camera, 1 for tracking camera and 2 for webcames. -->
  <arg name="frame_rate" default="30" />
  <arg name="width" default="1920" /> <!-- 1920 for main camera, 1280 for tracking camera --> 
  <arg name="height" default="1080" /> <!-- 1080 for main camera, 720 for tracking camera -->
  <arg name="input_format" default="NV12" /> <!-- format for the input stream from camera -->
  <arg name="output_format" default="RGB" /> <!-- format for the output stream -->
  <arg name="topic_name_prefix" default="camera" />
  <arg name="image_compress" default="true" /> 

  <node pkg="rb5_ros2_vision" exec="rb_camera_ocv_node" name="rb_camera_ocv_node" output="screen">
    <param name="camera_id" value="$(var camera_id)" />
    <param name="frame_rate" value="$(var frame_rate)" />
    <param name="width" value="$(var width)" />
    <param name="height" value="$(var height)" />
    <param name="input_format" value="$(var input_format)" />
    <param name="output_format" value="$(var output_format)" />
    <param name="topic_name" value="$(var topic_name_prefix)_$(var camera_id)" />
    <param name="image_compress" value="$(var image_compress)" />
  </node>
</launch>
